tests/vegalite/v5/test_theme.py

```diff
@@ -52,3 +53,665 @@ def custom_theme() -> dict[str, int]:
     assert is_color_hex(color_code) == valid
+
+
+def test_theme_config_typing(*, enable_mypy: bool = True) -> None:
+    """
+    Declares most of the default themes as ``ThemeConfig``.
+
+    Adding a typed argument to the signature ensures ``mypy`` checks this.
+    """
+    # ruff: noqa: F841
+
+    carbonwhite: ThemeConfig = {
+        "arc": {"fill": "#6929c4"},
+        "area": {"fill": "#6929c4"},
+        "axis": {
+            "grid": True,
+            "gridColor": "#e0e0e0",
+            "labelAngle": 0,
+            "labelColor": "#525252",
+            "labelFont": 'IBM Plex Sans Condensed, system-ui, -apple-system, BlinkMacSystemFont, ".SFNSText-Regular", sans-serif',
+            "labelFontSize": 12,
+            "labelFontWeight": 400,
+            "titleColor": "#161616",
+            "titleFontSize": 12,
+            "titleFontWeight": 600,
+        },
+        "axisX": {"titlePadding": 10},
+        "axisY": {"titlePadding": 2.5},
+        "background": "#ffffff",
+        "circle": {"fill": "#6929c4"},
+        "group": {"fill": "#ffffff"},
+        "path": {"stroke": "#6929c4"},
+        "range": {
+            "category": [
+                "#6929c4",
+                "#1192e8",
+                "#005d5d",
+                "#9f1853",
+                "#fa4d56",
+                "#570408",
+                "#198038",
+                "#002d9c",
+                "#ee538b",
+                "#b28600",
+                "#009d9a",
+                "#012749",
+                "#8a3800",
+                "#a56eff",
+            ],
+            "diverging": [
+                "#750e13",
+                "#a2191f",
+                "#da1e28",
+                "#fa4d56",
+                "#ff8389",
+                "#ffb3b8",
+                "#ffd7d9",
+                "#fff1f1",
+                "#e5f6ff",
+                "#bae6ff",
+                "#82cfff",
+                "#33b1ff",
+                "#1192e8",
+                "#0072c3",
+                "#00539a",
+                "#003a6d",
+            ],
+            "heatmap": [
+                "#f6f2ff",
+                "#e8daff",
+                "#d4bbff",
+                "#be95ff",
+                "#a56eff",
+                "#8a3ffc",
+                "#6929c4",
+                "#491d8b",
+                "#31135e",
+                "#1c0f30",
+            ],
+        },
+        "rect": {"fill": "#6929c4"},
+        "shape": {"stroke": "#6929c4"},
+        "style": {
+            "guide-label": {
+                "fill": "#525252",
+                "font": 'IBM Plex Sans,system-ui,-apple-system,BlinkMacSystemFont,".sfnstext-regular",sans-serif',
+                "fontWeight": 400,
+            },
+            "guide-title": {
+                "fill": "#525252",
+                "font": 'IBM Plex Sans,system-ui,-apple-system,BlinkMacSystemFont,".sfnstext-regular",sans-serif',
+                "fontWeight": 400,
+            },
+        },
```

Comment:
`"guide-(style|label)"` seem to have been skipped during schema gen, assuming the issue is the hyphen.

There are 4 hyphenated properties here:
https://github.com/vega/altair/blob/682b1695526d523b769f419b2dade1e32f5383de/altair/vegalite/v5/schema/vega-lite-schema.json#L28138-L28153

Reply:
@binste are you aware of this issue?

They appear in the docstring, but not in the signature for `core.StyleConfigIndex`

https://github.com/vega/altair/blob/682b1695526d523b769f419b2dade1e32f5383de/altair/vegalite/v5/schema/core.py#L22523-L22552

Reply:
Fixed in [`0bdfa72` (#3536)](https://github.com/vega/altair/pull/3536/commits/0bdfa72470551b2bc39e4adc76b7e989ccdf259b)